### PR TITLE
Implementing convolution using WGSL on WebGPU

### PR DESCRIPTION
**Pull request by Dp-Goog**
v1: Trying out convolution using WebGPU. Still looking into how to parallelize this work.
v2: Parallelize convolution by dispatching in work that is the size of the convolution array. Added a simple test that verifies that the IrHelper generated TestIR works properly by passing in an impulse pulse as an input.

### CODE REVIEW

File: src/experiments/webgpuaudio/main.js [hunk 1]
```diff
@@ -13,6 +13,10 @@ let toggleButton = null;
 let audioContext = null;
 let isPlaying = false;
 
+// Change this value to true to run tests and see test errors on JS console
+// during audio processing. Not intended for production, only for testing.
+let runTests = false;
```
Comment (hoch) [inline src/experiments/webgpuaudio/main.js:18]: How about having `TEST_MODE = false` in `constants.js` and use it by importing the file everywhere?

Comment (hoch) [reply src/experiments/webgpuaudio/main.js:18]: Then we don't have to pass it around via postMessage and alike.

File: src/experiments/webgpuaudio/test/test_processor.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
+import GPUProcessor from "../gpu-processor.js";
+import IRHelper from "../ir-helper.js"
+
+class TestProcessor {
+
+    constructor() {}
+    testConvolution = async() => {
```
Comment (hoch) [inline src/experiments/webgpuaudio/test/test_processor.js:7]: I was surprised that this syntax works, but the formal ES6 notation would be:
```js
constructor() {}

async testConvolution() {
...
}
```

File: src/experiments/webgpuaudio/main.js (hunk 1 above)
Comment (Dp-Goog) [reply src/experiments/webgpuaudio/main.js:18]: Good idea, done!

File: src/experiments/webgpuaudio/test/test_processor.js (hunk 2 above)
Comment (Dp-Goog) [reply src/experiments/webgpuaudio/test/test_processor.js:7]: Done!

**Review by hoch — APPROVED** [review]
LGTM

**Pull request merged**